test/integration/azureSpecialProperties.spec.ts

```diff
@@ -33,16 +33,26 @@ describe("auth validation", () => {
     const client = new AzureSpecialPropertiesClient(
       mockCredential,
       "1234-5678-9012-3456",
-      { allowInsecureConnection: true }
+      {
+        allowInsecureConnection: true,
+        credential: mockCredential,
+        credentialScopes: expectedScopes
+      }
     );
 
-    await client.apiVersionDefault.getMethodGlobalValid(responseStatusChecker);
+    let _response: FullOperationResponse;
+    await client.apiVersionDefault.getMethodGlobalValid({
+      ...responseStatusChecker,
+      onResponse: r => {
+        _response = r;
+      }
+    });
 
     // Validate auth header
-    // assert.equal(
-    //   result._response.request.headers.get("authorization"),
-    //   "Bearer test-token"
-    // );
+    assert.equal(
+      _response!.request.headers.get("authorization"),
+      "Bearer test-token"
+    );
```

Comment:
can we move the check to inside the callback?

Reply:
1. We are using this pattern across all test cases. 
2. I think it is more cleaner approach to have the asserts outside.

Reply:
TBH I usually worry when I see code extracting stuff from the callback, in general you don't have guarantees on when the callback is going to be called, so consuming a value set inside the callback is risky and can lead to race conditions. 

In this case it works because we know that by the time getMethodGlobalValid has resolved onResponse would've been called.

I'm not going to block on this but would also prefer to see the checks inside